src/Backend/WASMWrapper.js

```diff
@@ -29,36 +31,21 @@ export default class WASMWrapper {
         return this.#_wasm
     }
 
-    async getNames(values, func) {
-        const wasm = await this.wasm()
-        const getName = wasm.cwrap(func, 'number', ['number'])
-        return values.map(val => {
-            return {
-                id: val,
-                name: wasm.UTF8ToString(getName(val))
-            }
-        })
-    }
-
     /**
      * Fetches a list of libdedx programs and encapsulates them in the form of LibdedxEntity
      * @returns {LibdedxEntity[]} array of libdedx programs
      */
     async getPrograms() {
         const wasm = await this.wasm()
 
-        const buf = wasm._malloc(this.#programsSize * Int32Array.BYTES_PER_ELEMENT)
-        const heap = new Uint32Array(wasm.HEAP32.buffer, buf, this.#programsSize)
+        const [progPtr, programs] = this._allocateI32(wasm, this.#programsSize)
 
-        wasm.ccall("dedx_get_program_list", null, ['number'], [heap.byteOffset])
+        wasm.ccall("dedx_fill_program_list", null, ['number'], [programs.byteOffset])
+        const result = Array.from(programs.subarray(0, programs.indexOf(-1)))
```

Comment:
Why using `programs.indexOf(-1)` ? Is it connected with the fact that list of programs from libdedx is terminated with -1 ?

Reply:
That is the exact reason. Since we cannot assume the size the incomming array has i allocate more memory than is needed and then seek for the terminator value which in this case is -1

Reply:
```suggestion
        // libdedx stores the list on programs in an array of much larger size than its needed
        // list of useful program is terminated with `-1` entry
        // we use it here to extract list of of programs (excluding `-1` terminator)
        const result = Array.from(programs.subarray(0, programs.indexOf(-1)))
```